Sources/Basic/Lock.swift

```diff
@@ -58,10 +62,34 @@ public final class FileLock {
         self.cachePath = cachePath
     }
 
-    /// Try to aquire a lock. This method will block until lock the already aquired by other process.
+    /// Registers signal handler for SIGUSR1 and raises the signal
+    /// after the given number of seconds.
+    private func setupTimeout(seconds: Double) {
+        // Reset the timer indicator.
+        timerTicked = false
+        // Register signal handler to avoid process termination.
+        SignalManager.shared.register(.usr1) {}
+        // Raise the signal when we reach timeout.
+        DispatchQueue.global().asyncAfter(deadline: .now() + seconds) {
+            // Tick the timer.
+            self.timerTicked = true
+            // Raise the signal.
+            SignalManager.shared.raise(.usr1)
+        }
+    }
```

Comment:
This won't work correctly if multiple threads ever try to lock with a timeout simultaneously... maybe we should just document that is unsupported (i.e. that FileLock should only be used from one thread)?